libgalois/include/katana/PODResizeableArray.h

```diff
@@ -138,7 +140,7 @@ class PODResizeableArray {
         data_ = NULL;
         capacity_ = 0;
       }
-    } else if (size_ < capacity_) {
+    } else if ((size_ < capacity_) && (size_ < kMinNonZeroCapacity)) {
```

Comment:
We don't want to realloc if `size >= kMinNonZeroCapacity`?
 
and below, regarding the statement `capacity_ = size_`. If `size < kMinNonZeroCapacity`, capacity_ could be less than kMinNonZeroCapacity (and non zero), which seems non-intuitive.

Reply:
Yes, sorry. That was a mistake. It should be `size_ >= kMinNonZeroCapacity`